web/backend/src/Server.ts

```diff
@@ -177,46 +186,49 @@ app.post('/api/logout', (req, res) => {
 // the react. This endpoint serves to send to the client (actually to react)
 // the information of the current user.
 app.get('/api/personal_info', (req, res) => {
-  res.set('Access-Control-Allow-Origin', '*');
-  if (req.session.userid) {
-    res.json({
-      sciper: req.session.userid,
-      lastname: req.session.lastname,
-      firstname: req.session.firstname,
-      role: req.session.role,
-      islogged: true,
-    });
-  } else {
-    res.json({
-      sciper: 0,
-      lastname: '',
-      firstname: '',
-      role: '',
-      islogged: false,
-    });
-  }
+  const m = new Map<String, Array<String>>();
+  enf.getFilteredPolicy(0, String(req.session.userid)).then((list) => {
+    for (let i = 0; i < list.length; i += 1) {
+      for (let j = 1; j < list[0].length; j += +2) {
+        console.log(list[i][j]);
+        if (m.has(list[i][j])) {
+          m.get(list[i][j])?.push(list[i][j + 1]);
```

Comment:
Even if I check `m.has` I still have the error (Object is possibly 'undefined') This why I used `?`. However it can be a good way to test the result of `m.get`. I was just wondering how it is better than using `?`